Update angular quickstart (#139)
teamhanko/docs: quickstarts/frontend/angular.mdx

```

```bash bun
bun install -g @angular/cli
ng new project-name --style=css --routing --standalone --defaults
```

```bash yarn
yarn global add @angular/cli
ng new project-name --style=css --routing --standalone --defaults
``` 

</CodeGroup>

<Note>
  This quickstarts examples are based on the Server Routing and App Engine APIs.
  Make sure while creating your angular project you accept (press 'y') the routing and App Engine.
</Note>

## Install `@teamhanko/hanko-elements`

Once you've initialized your Angular app, installing hanko-elements provides you with access to the prebuilt components: `hanko-auth` and `hanko-profile`.

<CodeGroup>
```bash npm
cd project-name
npm install @teamhanko/hanko-elements
``` 

```bash pnpm
cd project-name
pnpm add @teamhanko/hanko-elements
```

```bash bun
cd project-name
bun add @teamhanko/hanko-elements
```

```bash yarn
cd project-name
yarn add @teamhanko/hanko-elements
```
</CodeGroup>

## Setup your Hanko project
Go to the [Hanko console](https://cloud.hanko.io/) and [create a project for this application.](/setup-hanko-cloud)

<Note>
  During creation make sure to input the URL you will be developing on as the `APP URL`.

>   (Most likely http://localhost:5173/) 

Reviewer: Isn't the default port for Angular `4200`?

[For more information please refer to the Auth Component Page.](/guides/hanko-elements/auth-component)

<Tabs>
<Tab title='hanko-auth/hanko-auth.component.html'>
```html
<hanko-auth (onSessionCreated)="redirectAfterLogin()" />
```

</Tab>
<Tab title='hanko-auth/hanko-auth.component.ts'>
```ts
//CUSTOM_ELEMENTS_SCHEMA lets you use the Hanko Elements
import { Component, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { Router } from '@angular/router';
//function to register hanko element
import { register } from "@teamhanko/hanko-elements";

import { environment } from '../../environments';

@Component({
  //Selector you will use in the pages html files
  selector: 'app-hanko-auth',
  imports: [],
  templateUrl: './hanko-auth.component.html',
  styleUrl: './hanko-auth.component.css',
  schemas: [CUSTOM_ELEMENTS_SCHEMA],
})
export class HankoAuthComponent {
  hankoApi = environment.HANKO_API_URL;

  constructor(private router: Router) {
    register(this.hankoApi).catch((error) => {
      // handle error
    });
  }

  redirectAfterLogin() {
    //Succesfully logged in, redirect to any path you want.
    this.router.navigate(['/dashboard']);
  }
}
```
</Tab>
</Tabs>

> Now Import the HankoAuth component into the Home Page you previously made.

Reviewer: ```suggestion
Now Import the HankoAuth component into the home page you previously made.
```

<Tabs>
<Tab title='home/home.page.html'>
```html
<!-- This is the selector from the hanko-auth.component.ts file -->
<app-hanko-auth/>
```

</Tab>
<Tab title='home/home.page.ts'>
```ts
  import { Component } from '@angular/core';

  import { HankoAuthComponent } from '../hanko-auth/hanko-auth.component';

  @Component({
    selector: 'app-home',
    imports: [HankoAuthComponent],//Import the component into your page
    templateUrl: './home.page.html',
    styleUrl: './home.page.css'
  })
  export class HomePage {

  }
```
</Tab>
</Tabs>

### Hanko profile

After setting up the HankoAuth let's set up the `HankoProfile` component to create a interface where users can.\

> manage their `Email Addresses` and credentials.

Reviewer: ```suggestion
After setting up the HankoAuth let's set up the `HankoProfile` component to create a interface where users can manage their `Email Addresses` and credentials.
```

[For more information please refer to the Profile Component Page.](/guides/hanko-elements/profile-component)

<Tabs>
<Tab title='hanko-profile/hanko-profile.component.html'>
```html
<hanko-profile/>
```
</Tab>

<Tab title='hanko-profile/hanko-profile.component.ts'>
```ts
//CUSTOM_ELEMENTS_SCHEMA lets you use the Hanko Elements
import { Component , CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
//function to register hanko element
import { register } from '@teamhanko/hanko-elements';

import { environment } from '../../environments';

@Component({
  //Selector you will use in the pages html files
  selector: 'app-hanko-profile',
  imports: [],
  templateUrl: './hanko-profile.component.html',
  styleUrl: './hanko-profile.component.css',
  schemas: [CUSTOM_ELEMENTS_SCHEMA]
})
export class HankoProfileComponent {
  hankoApi = environment.HANKO_API_URL;

  constructor() {
    register(this.hankoApi).catch((error) => {
      // handle error
    });
  }
}
```
</Tab>
</Tabs>

> Now Import the HankoAuth component into the Dashboard Page you previously made.

Reviewer: ```suggestion
Now Import the HankoAuth component into the dashboard page you previously made.
```

    templateUrl: './dashboard.page.html',
    styleUrl: './dashboard.page.css'
  })
  export class DashboardPage {

  }
```
</Tab>
</Tabs>

### Setting up routes

Remove everything in the `app.component.html` file and add a router outlet.

```html app.component.html
<router-outlet></router-outlet>
```

Now inport both pages into the `app.routes.ts` file and setup your routes.
```ts app.routes.ts
import { Routes } from '@angular/router';

import { HomePage } from './home/home.page';
import { DashboardPage } from './dashboard/dashboard.page';

export const routes: Routes = [
  {
    path: '',
    component: HomePage,
  },
  {
    path: 'dashboard',
    component: DashboardPage,
  },
];
```

By now you should be able to go to `/` to see the `<HankoAuth>`, and to `/dashboard` to see the `<HankoProfile>`.

They should look something like this👇
<Frame>
  <div style={{justifyContent: 'center', alignItems: 'center', display: 'flex', backgroundColor: 'white'}}>
    <img src="/images/fullstack-guide/next-one.png" alt="sign up" width="500"/>
    <img src="/images/fullstack-guide/next-two.png" alt="profile page" width="500" />
  </div>
</Frame>

<Note>

>   use the command `ng serve` to start the server.

Reviewer: ```suggestion
  Use the command `ng serve` to start the server.
```

</Note>

## Implement logout functionality

You can use `@teamhanko/hanko-elements` to easily log users out. Here we will make a logout button component that you can use anywhere.

Create the LogoutButton component with this command.
```bash
ng generate component LogoutButton --standalone
```

> Now let's set up the HTML and TS of the LogoutButton component.

Reviewer: ```suggestion
Now let's set up the LogoutButton component.
```

```ts 
import { inject } from '@angular/core';
import { CanActivateFn, Router } from '@angular/router';

export const authGuard: CanActivateFn = async (route, state) => {
  const router = inject(Router);
  const backendUrl = "http://localhost:5001";//change this to the url of your running hanko backend

  const sessionToken = isBrowser() ? getCookie('hanko') : null;
  if (!sessionToken) throw new Error('No session token found.');

  try {
    const response = await fetch(`${backendUrl}/validate`, {
      credentials: "include",
    });

    if (!response.ok) throw new Error('Session validation failed');

    return true;
  } catch (error) {
    console.error('AuthGuard error:', error);
    return router.parseUrl('/');// url to return the user to if they are not authenticated
  }
};

function getCookie(name: string): string | null {
  if (typeof document === 'undefined') return null;
  const match = document.cookie.match(new RegExp(`(^| )${name}=([^;]+)`));
  return match ? match[2] : null;
}

function isBrowser(): boolean {
  return typeof window !== 'undefined' && typeof document !== 'undefined';
}
```

> Now to use this authentication guard you have to add it to your `app.routes.ts` file.

Reviewer: ```suggestion
Add this authentication guard to the routes, in the `app.routes.ts`, which should only be accessible when the user is logged in.
```

```ts app.routes.ts
import { Routes } from '@angular/router';

import { HomePage } from './home/home.page';
import { DashboardPage } from './dashboard/dashboard.page';

//Import the auth guard
import { authGuard } from './auth.guard';

export const routes: Routes = [
  {
    path: '',
    component: HomePage,
  },
  {
    path: 'dashboard',
    component: DashboardPage,
    canActivate: [authGuard]//secure the /dashboard path with the guard
  },
];
```

> To verify that it works, Logout on your app and go to /dashboard, you should get redirected back!

Reviewer: ```suggestion
To verify that it works, logout on your app and go to /dashboard, you should get redirected back!
```

## Getting user data

Let's use the Hanko SDK to get user data to display on our dashboard page.

> For the User Information we will use [hanko.GetUser()](https://teamhanko.github.io/hanko/jsdoc/hanko-frontend-sdk/index.html#getting-the-user-object)

Reviewer: ```suggestion
For the user information we will use [hanko.GetUser()](https://teamhanko.github.io/hanko/jsdoc/hanko-frontend-sdk/index.html#getting-the-user-object)
```

> Lets update our dashboard page html and ts files.

Reviewer: ```suggestion
Lets update our dashboard page.
```
